### ISSUE TITLE
Not able to fetch latest available  version from store

### ISSUE THREAD

**Issue by chauhan-saurabh** [GantMan/react-native-siren#17]
I just pushed an update to the app store and it's almost 12 hours the plugin is returning the older version and when I am opening the store it shows an update available.

**Comment by fernandocfbf** [comment]
same problem here!
Could you please provide how long it takes to update?
Thank you in advance!

**Comment by GantMan** [comment]
You're calling "Perform check" and it's not working?

DO you see the new version when you look at https://apps.apple.com/app/id${appId}?mt=8?

**Comment by fernandocfbf** [comment]
Yes, I can see the new version on the store, but the library is getting the wrong version number

**Comment by GantMan** [comment]
That's wild.  Maybe it's some kind of cache?  Did you remove the app from memory?

**Comment by testdevappapptest** [comment]
Hi @GantMan , I am also facing a similar issue. After App Store deployment, the TestFlight build is not getting the latest AppStore version. In debug mode, I am getting the latest version. It took around 24 hours for the TestFlight build to get the new version.
Is there any way to reduce the delay?

**Comment by penji2** [comment]
Same here, but in my case app deployed in one region only. I think I found root cause of the problem. It seems like passing country code as query parameter is causing trouble. So if you query this: `https://itunes.apple.com/lookup?bundleId=yourBundleId&country=yourAppsRegion` the API will response with older version of the app info. If you wait for about 16-22h (not sure about exact time) it will be refreshed, and you'll get correct response. I found out, that when you provide country code as part of URL, problem doesn't appear. So if you go like: `https://itunes.apple.com/yourAppsRegion/lookup?bundleId=yourBundleId`. I can create a PR fixing that. Let me know if its ok.

**Comment by wichan7** [comment]
There is a caching issue with the iTunes Lookup API.
It is unclear whether the problem is caused by client-side or server-side caching,
but the API functions correctly when we bypass the cache by appending a query parameter such as /lookup?ts=1747401120293

Please check PR #25
